rocAL_pybind/amd/rocal/plugin/generic.py

```diff
@@ -130,46 +121,24 @@ def __next__(self):
             self.output_list = []
             for i in range(len(self.output_tensor_list)):
                 self.dimensions = self.output_tensor_list[i].dimensions()
-                if self.device == "cpu":
-                    self.dtype = self.output_tensor_list[i].dtype()
-                    self.output = np.empty(self.dimensions, dtype=self.dtype)
-                    self.labels = np.empty(self.labels_size, dtype="int32")
-                else:
-                    self.dtype = self.output_tensor_list[i].dtype()
-                    with cp.cuda.Device(device=self.device_id):
-                        self.output = cp.empty(
-                            self.dimensions, dtype=self.dtype)
-                        self.labels = cp.empty(
-                            self.labels_size, dtype="int32")
-
-                if self.device == "cpu":
-                    self.output_tensor_list[i].copy_data(self.output)
-                else:
-                    self.output_tensor_list[i].copy_data(self.output.data.ptr)
+                self.dtype = self.output_tensor_list[i].dtype()
+                self.output = np.empty(self.dimensions, dtype=self.dtype)
+                self.labels = np.empty(self.labels_size, dtype="int32")
```

Comment:
The initialization of self.labels is only required for one hot encoding case. For other cases, its returned as numpy arrays. Can we move this line to L140?